ada_feeding_perception/ada_feeding_perception/food_on_fork.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+import rclpy
+from rclpy.node import Node
+from rclpy.executors import MultiThreadedExecutor
+
+from sensor_msgs.msg import Image
+from std_msgs.msg import Float32
+
+from cv_bridge import CvBridge, CvBridgeError
+import numpy as np
+import joblib
+
+
+class FoodOnFork(Node):
+    """
+    Class FoodOnFork inherits from the Node class. This is a node that subscribes to the
+    raw depth image and publishes to a topic `food_on_fork`. The published messages are of type
+    Float32. The published messages return the probability of the presence of food on the fork.
+    """
+    def __init__(self):
+        """
+        Initialize FoodOnFork node
+        """
+        super().__init__('food_on_fork')
+
+        # depth topic subscription
+        self.subscription_depth = self.create_subscription(
+            Image,
+            'camera/depth/image_rect_raw',
+            self.listener_callback_depth,
+            1
+        )
+        self.subscription_depth
+
+        # publisher for FoF vs. no FoF
+        self.publisher_depth = self.create_publisher(
+            Float32,
+            'food_on_fork',
+            1
+        )
+
+        # cv bridge
+        self.bridge = CvBridge()
+
+        # initialize the model
+        self.model = joblib.load("/home/atharva2/atharvak_node_ws/src/ada_feeding/ada_feeding_perception/ada_feeding_perception/logistic_reg_model.pkl")
+
+    def listener_callback_depth(self, depth_img_msg):
+        """
+        Calculates and publishes probability indicating whether the provided depth image has food on fork
+
+        Parameters:
+            depth_img_msg: depth image message from the subscriber
+        """
+        depth_img = None
+        try:
+            depth_img = self.bridge.imgmsg_to_cv2(depth_img_msg, "passthrough")
+        except CvBridgeError as e:
+            print(e)
+
+        if depth_img is not None:
+            num_pixels = self.food_on_fork_num_pixels(depth_img)
+            # print("num pixels: ", num_pixels)
+            prediction = self.predict_food_on_fork(num_pixels)
+            # print("prediction: ", prediction)
+            float32msg = Float32()
+            float32msg.data = prediction
+            self.publisher_depth.publish(float32msg)
+
+    def food_on_fork_num_pixels(self, depth_img,
+                                left_top_corner=(297, 248),
+                                right_bottom_corner=(422, 332),
+                                min_dist=(330 - 20),
+                                max_dist=(330 + 40)) -> int:
```

Comment:
All four of these should be passed in as rosparams instead of function parameters. That way, we can change the parameters without having to modify the code.